pkg/kn/core/root.go

```diff
@@ -87,7 +87,8 @@ func NewDefaultKnCommandWithArgs(rootCmd *cobra.Command,
 				fmt.Fprintf(rootCmd.OutOrStderr(), "Error: unknown command '%s' \nRun 'kn --help' for usage.\n", args[1])
 				os.Exit(1)
 			}
-		} else if foundCmd.HasSubCommands() {
+		}
+		if foundCmd.HasSubCommands() {
```

Comment:
This can either be kept as an `else if` and checking would need to be done in the `if` block so an error is still returned if the command is invalid, or it can be checked each time as its own block.

Reply:
Thanks. I actually also had a fix in my pending PR for allowing extending all subcommands groups. But thanks for submitting this. I’ll merge mine.